pyannote/audio/interactive/diarization/recipehelper.py

```diff
@@ -0,0 +1,272 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2021-2022 CNRS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Jim Petiot
+# Hervé Bredin
+
+import random
+from copy import deepcopy
+from datetime import datetime
+from itertools import groupby
+from pathlib import Path
+from typing import Dict, Iterable, List
+
+import numpy as np
+import torch
+from pyannote.core import Annotation, Segment
+from scipy.spatial.distance import cdist
+
+from pyannote.audio import Audio, Inference, Pipeline
+
+from ..common.utils import AudioForProdigy, get_audio_spans, get_chunks
+
+
+class RecipeHelper:
+    def __init__(self, embedding="pyannote/embedding"):
+        self.inference = Inference(embedding, window="whole")
+        dim = self.inference.model.introspection.dimension
+        self.speakers = np.array(
+            [], dtype=[("name", "U100"), ("embedding", "f4", dim), ("nb", "i4")]
+        )
+        self.buffer = {}
+
+    def on_exit(self, controller):
+        now = datetime.now()
+        date_time = now.strftime("%d-%m-%Y")
+        name = "embeddings_" + date_time
+        np.save(name, self.speakers)
+
+    def update(self, answers):
+        eg = answers[0]
```

Comment:
This assumes that answers only contain one answer (because `batch_size = 1`), right? 
What if we want to change that for some reason in the future? 
It would be better to loop `for eg in answers` here.